### PR TITLE
feat: Add `array_destructuring` as option for `trailing_comma_in_multiline`

### PR DESCRIPTION
**Pull request by SpacePossum**
@ see https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/7247

Currently the option `arguments` already supports destruction when using the older `list()` syntax, but there was no support yet for the newer `[] = ` syntax.

### CODE REVIEW

File: src/Fixer/ControlStructure/TrailingCommaInMultilineFixer.php [hunk 1]
```diff
@@ -130,22 +132,26 @@ protected function createConfigurationDefinition(): FixerConfigurationResolverIn
 
     protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
     {
-        $fixArrays = \in_array(self::ELEMENTS_ARRAYS, $this->configuration['elements'], true);
-        $fixArguments = \in_array(self::ELEMENTS_ARGUMENTS, $this->configuration['elements'], true);
-        $fixParameters = \PHP_VERSION_ID >= 8_00_00 && \in_array(self::ELEMENTS_PARAMETERS, $this->configuration['elements'], true); // @TODO: drop condition when PHP 8.0+ is required
-        $fixMatch = \PHP_VERSION_ID >= 8_00_00 && \in_array(self::MATCH_EXPRESSIONS, $this->configuration['elements'], true); // @TODO: drop condition when PHP 8.0+ is required
+        $configuredElements = $this->configuration['elements'];
```
Comment (SpacePossum) [inline src/Fixer/ControlStructure/TrailingCommaInMultilineFixer.php:135]: `diff` looks rather bad, however it is just the extraction `$configuredElements = $this->configuration['elements'];` to reduce the complexity when trying to read the code, and than simplify the checks below it.

File: doc/rules/control_structure/trailing_comma_in_multiline.rst [hunk 2]
```diff
@@ -2,8 +2,8 @@
 Rule ``trailing_comma_in_multiline``
 ====================================
 
-Multi-line arrays, arguments list, parameters list and ``match`` expressions
-must have a trailing comma.
+Multi-line arrays, arguments list, parameters list,  ``match`` and array
```
Comment (kubawerlos) [inline doc/rules/control_structure/trailing_comma_in_multiline.rst:5]: What about sorting the elements in `setAllowedValues` call in `TrailingCommaInMultilineFixer`, so these here are in the same order as in line 26?

Comment (SpacePossum) [reply doc/rules/control_structure/trailing_comma_in_multiline.rst:5]: Thanks, I think I've done what you are suggesting, let me know if I misunderstood and I'll adjust accordingly

**Comment by SpacePossum** [comment]
rebased on latest and forced without other changes

File: doc/rules/control_structure/trailing_comma_in_multiline.rst (hunk 2 above)
Comment (kubawerlos) [reply doc/rules/control_structure/trailing_comma_in_multiline.rst:5]: I was thinking to sort them alphabetically by value, so arguments option goes before both array options and params option is last

Comment (SpacePossum) [reply doc/rules/control_structure/trailing_comma_in_multiline.rst]: thanks updated :+1:

**Review by kubawerlos — APPROVED** [review]

**Comment by kubawerlos** [comment]
Thank you @SpacePossum

**Pull request merged**